src/main/java/com/sprint/mission/discodeit/service/jcf/JCFMessageService.java

```diff
@@ -6,27 +6,45 @@
 
 public class JCFMessageService implements MessageService {
     private final Map<UUID, Message> data = new HashMap<>();
-    private JCFUserService userService;
-    private JCFChannelService channelService;
 
-    public JCFMessageService(JCFUserService userService, JCFChannelService channelService) {
-        this.userService = userService;
-        this.channelService = channelService;
+    public JCFMessageService() {}
+
+    @Override
+    public Message create(String content, UUID channelId, UUID authorId) {
+        Message message = new Message(content, channelId, authorId);
+        this.data.put(message.getId(), message);
+
+        return message;
     }
 
-    public void createMessage(Message message) {
-        if(userService.getUser(message.getUserId()) != null && channelService.getChannel(message.getChannelId()) != null)
-        {
-            data.put(message.getId(), message);
-        }
-        else System.out.println("User not found");
+    @Override
+    public Message find(UUID messageId) {
+        Message messageNullable = this.data.get(messageId);
+
+        return Optional.ofNullable(messageNullable)
+                .orElseThrow(() -> new NoSuchElementException("Message with id " + messageId + " not found"));
```

Comment:
Optional은 메서드의 반환 타입으로 사용되는 것이 일반적인데, 여기서는 중간 처리 과정에서만 사용되고 있습니다. 이런 경우 Optional을 사용하는 목적이 희석됩니다.

1) 그냥 널 체크를 쓰시던가
```suggestion
Message message = this.data.get(messageId);
    if (message == null) {
        throw new NoSuchElementException("Message with id " + messageId + " not found");
    }
    return message;
```

인터페이스 상에서 Optional<Message> 반환을 허용하면 

```suggestion
        @Override
public Optional<Message> find(UUID messageId) {
    return Optional.ofNullable(this.data.get(messageId));
}
```

이게 Optional을 정확하게 쓰는 방식에 가깝습니다. 자바의 고급 기능을 활용하려고 시도한 건 아주 좋아요. 다만, 아직 익숙치 않다보니 어색하게 느껴집니다. 앞으로 배우면 되니 걱정 말고 계속 시도해요 👍